app/src/main/java/chat/rocket/android/chatroom/ui/ChatRoomFragment.kt

```diff
@@ -102,19 +121,40 @@ class ChatRoomFragment : Fragment(), ChatRoomView {
         }
     }
 
+    override fun uploadFile(uri: Uri) {
+        var fileName: String? = null
+        var mimeType: String? = null
+        var fileRealPath: String? = null
+        activity?.apply {
+            fileName = uri.getFileName(this)
```

Comment:
yout can move the declaration down here, and use immutable `val`
```kotlin
val filename = uri.getFileName(this)
```

Maybe move all this to the presenter, just call `presenter.uploadFile(uri, message)` and create an interactor for retrieving the information from the `Uri`

It is better for handling errors, and calling `view.showInvalidFile()` for example...